Bottom: c441b59ea669cdce6f3eda17dd06cd3cedb46376
Top:    259150d5ba4a31c22d5f4e23c73dd3f9211ed7ae
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 17:26:25 +0000

Refresh of hid-gcode-don-t-mix-g_-allocat

---

diff --git a/src/file.c b/src/file.c
index f71d1b0..dd543aa 100644
--- a/src/file.c
+++ b/src/file.c
@@ -1405,7 +1405,6 @@ ParseLibraryTree (void)
   printf("Leaving ParseLibraryTree, found %d footprints.\n", n_footprints);
 #endif
 
-  free (libpaths);
   return n_footprints;
 }
 
@@ -1539,11 +1538,10 @@ ReadNetlist (char *filename)
   LibraryEntryTypePtr entry;
   int i, j, lines, kind;
   bool continued;
-  bool used_popen = false;
-  int retval = 0;
+  int used_popen = 0;
 
   if (!filename)
-    return 1;			/* nothing to do */
+    return (1);			/* nothing to do */
 
   Message (_("Importing PCB netlist %s\n"), filename);
 
@@ -1558,7 +1556,7 @@ ReadNetlist (char *filename)
     }
   else
     {
-      used_popen = true;
+      used_popen = 1;
       free (command);
       command = EvaluateFilename (Settings.RatCommand,
 				  Settings.RatPath, filename, NULL);
@@ -1567,7 +1565,7 @@ ReadNetlist (char *filename)
       if (*command == '\0' || (fp = popen (command, "r")) == NULL)
 	{
 	  PopenErrorMessage (command);
-	  return 1;
+	  return (1);
 	}
     }
   lines = 0;
@@ -1638,14 +1636,15 @@ ReadNetlist (char *filename)
   if (!lines)
     {
       Message (_("Empty netlist file!\n"));
-      retval = 1;
+      pclose (fp);
+      return (1);
     }
   if (used_popen)
     pclose (fp);
   else
     fclose (fp);
   sort_netlist ();
-  return retval;
+  return (0);
 }
 
 static int ReadEdifNetlist (char *filename);
diff --git a/src/hid/common/hidinit.c b/src/hid/common/hidinit.c
index 3695f7d..0bdd395 100644
--- a/src/hid/common/hidinit.c
+++ b/src/hid/common/hidinit.c
@@ -117,7 +117,6 @@ hid_load_dir (char *dirname)
       free (path);
     }
   free (dirname);
-  closedir (dir);
 }
 
 void
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index b08d90d..9d88086 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -381,7 +381,7 @@ nogui_prompt_for (const char *msg, const char *default_string)
   if (answer == NULL)
     return strdup ((default_string != NULL) ? default_string : "");
   else
-    return answer;
+    return strdup (answer);
 }
 
 /* FIXME - this could use some enhancement to actually use the other
@@ -402,7 +402,7 @@ nogui_fileselect (const char *title, const char *descr,
   if (answer == NULL)
     return (default_file != NULL) ? strdup (default_file) : NULL;
   else
-    return answer;
+    return strdup (answer);
 }
 
 static int
diff --git a/src/hid/gtk/ghid-layer-selector.c b/src/hid/gtk/ghid-layer-selector.c
index 2b04063..432e57e 100644
--- a/src/hid/gtk/ghid-layer-selector.c
+++ b/src/hid/gtk/ghid-layer-selector.c
@@ -643,7 +643,7 @@ ghid_layer_selector_add_layer (GHidLayerSelector *ls,
       free_ldata (ls, new_layer);
     }
 
-  new_layer = g_malloc (sizeof (*new_layer));
+  new_layer = malloc (sizeof (*new_layer));
 
   gtk_list_store_set (ls->list_store,
                       &iter,
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index b58698e..6ab4993 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2117,7 +2117,6 @@ hid_gtk_init ()
   free (tmps);
 #undef REST_OF_PATH
   printf ("\"Share\" installation path is \"%s\"\n", share_dir);
-  free (share_dir);
 #endif
 
   memset (&ghid_hid, 0, sizeof (HID));
